### PR TITLE
Feature use sass

### PR DESCRIPTION
**Pull request by doubleO8**
Using Sass (https://sass-lang.com/) for CSS handling.

This pull request is more or less asking if you guys would be interested in using sass for managing and creating CSS files for the UI.
Using this approach make customisation and changes easier by using variables/constants and having CSS compiled and minified without further interactions. Creating "_themes_" is also rather easy with this.

For demonstration purposes I added a new CSS file `main_and_ui.css` which contain the two main CSS contents and could thusly be used to require one HTTP request less in `main.html`.

### CODE REVIEW

File: motioneye/static/css/_variables.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/* shared variables and constants */
+$body-default-color: #dddddd;
+$body-default-background: #212121;
```
Comment (cclauss) [inline motioneye/static/css/_variables.scss:3]: ⛔ Is GitHub wanting a carriage return at the end of the file.

File: motioneye/static/css/main.scss [hunk 2]
```diff
@@ -0,0 +1,2 @@
+@import '_variables';
+@import '_main';
```
Comment (cclauss) [inline motioneye/static/css/main.scss:2]: ⛔

**Review by cclauss — CHANGES_REQUESTED** [review]

**Comment by MichaIng** [comment]
Many thanks for your PR.

I'm not sure whether we want this in general or now in particular, due to the other larger changes under the hood (Python 3, script refactoring, i18n, ...).

Minifying and merging CSS can be done easily without SASS, also CSS variables are not a SASS features but can be used in plain CSS, which implies the ability for theming, given that only variables need to be adjusted. So currently I'm not aware of a particular feature request or enhancement which requires SASS.

The main problem I see is that we break the ability to simply download/clone and run this repo, but it requires SASS pre-processing first to work. Of course we could have SASS sources and compiled CSS both in the repo, but this again doubles code and adds the need to have a compiling step on every CSS change. Additional abstraction and processing layers are generally an additional burden and point of failure (including understanding, bugs, up to security issues) in the required pre-processing tools and dependencies. It only makes sense when one requires the additional features, or when it allows to simplify code significantly, now, not somewhere in the future, IMHO.

But this is my personal opinion only, open for others 🙂.

**Comment by zagrim** [comment]
One thing SASS (or LESS) provides over plain CSS is nesting, which I've found out to improve readability and somewhat reduce repetition (and thus improve maintainability) of stylesheets. This hasn't been used in the changes in this PR though (if it would, making sure the .scss files match the original .css would be difficult). That being said, I don't think it makes a huge difference but the stylesheets are complex/long enough for that to make some difference. Perhaps not just right now if there's no planned UI rework to be done.

> The main problem I see is that we break the ability to simply download/clone this repo and run motionEye, but it first requires SASS pre-processing to work.

I don't know how often that is being done, but if it was to be done by any non-developer who's not well versed with command line stuff, that might very well end up being a blocker.

**Comment by MichaIng** [comment]
Jep, that nesting is nice, but whether this would really reduce the amount of lines that much is another question. AFAIK there are simply many different classes used. Before using SASS only in the hope it reduces code lines somewhere in the future, I'd better go through the CSS now and check for obsolete styles and possibly smarter/modern solutions for layout issues which are currently addressed with possibly complicated or legacy methods. With cleaned up CSS it is also easier to apply SASS enhancements directly when adding support for it.

And generally, adding SASS usage should then come along with the related code enhancements directly. Else we have a feature which is not used with a chance that it is never used 😉.

> Perhaps not just right now if there's no planned UI rework to be done.

👍, if someone does a larger UI rework and finds SASS to make it easier, then it'd be fully open to have it implemented together with the UI rework. But as standalone change I see only downsides 🙂.

**Pull request closed without merging**